solo-chains/runtime/dancelight/src/tests/inbound_queue_tests/mock.rs

```diff
@@ -0,0 +1,347 @@
+        true
+    }
+
+    fn process_message(_channel: Channel, _envelope: Envelope) -> Result<(), DispatchError> {
+        panic!("DummySuffix::process_message shouldn't be called");
+    }
+}
+
+impl snowbridge_pallet_inbound_queue::Config for Test {
+    type RuntimeEvent = RuntimeEvent;
+    type Verifier = MockVerifier;
+    type Token = Balances;
+    type XcmSender = MockXcmSender;
+    type WeightInfo = ();
+    type GatewayAddress = GatewayAddress;
+    type MessageConverter = MessageToXcm<
+        CreateAssetCall,
+        CreateAssetDeposit,
+        InboundQueuePalletInstance,
+        AccountId,
+        Balance,
+    >;
+    type PricingParameters = Parameters;
+    type ChannelLookup = MockChannelLookup;
+    #[cfg(feature = "runtime-benchmarks")]
+    type Helper = Test;
+    type WeightToFee = IdentityFee<u128>;
+    type LengthToFee = IdentityFee<u128>;
+    type MaxMessageSize = ConstU32<1024>;
+    type AssetTransactor = SuccessfulTransactor;
+    type MessageProcessor = (
+        DummyPrefix,
+        XcmMessageProcessor<Test>,
+        SymbioticMessageProcessor<Test>,
+        DummySuffix,
+    );
+}
+
+pub struct ValidatorIdOf;
+impl sp_runtime::traits::Convert<AccountId, Option<AccountId>> for ValidatorIdOf {
+    fn convert(a: AccountId) -> Option<AccountId> {
+        Some(a)
+    }
+}
+
+pub struct DummyValidatorRegistration;
+
+impl ValidatorRegistration<AccountId> for DummyValidatorRegistration {
+    fn is_registered(_id: &AccountId) -> bool {
+        true
+    }
+}
+
+parameter_types! {
+
+    pub const MaxWhitelistedValidators: u32 = 100;
+    pub const MaxExternalValidators: u32 = 100;
+    pub const SessionsPerEra: SessionIndex = runtime_common::prod_or_fast!(6, 3);
+}
+
+pub struct DummyUnixTime;
+
+impl UnixTime for DummyUnixTime {
+    fn now() -> Duration {
+        Duration::default()
+    }
+}
+
+impl pallet_external_validators::Config for Test {
+    type RuntimeEvent = RuntimeEvent;
+    type UpdateOrigin = EnsureRoot<AccountId>;
+    type HistoryDepth = ConstU32<84>;
+    type MaxWhitelistedValidators = MaxWhitelistedValidators;
+    type MaxExternalValidators = MaxExternalValidators;
+    type ValidatorId = AccountId;
+    type ValidatorIdOf = ValidatorIdOf;
+    type ValidatorRegistration = DummyValidatorRegistration;
+    type UnixTime = DummyUnixTime;
+    type SessionsPerEra = SessionsPerEra;
+    type OnEraStart = ();
+    type OnEraEnd = ();
+    type WeightInfo = ();
+    #[cfg(feature = "runtime-benchmarks")]
+    type Currency = Balances;
+}
+
+pub fn setup() {
+    System::set_block_number(1);
+    Balances::mint_into(
+        &sibling_sovereign_account::<Test>(ASSET_HUB_PARAID.into()),
+        InitialFund::get(),
+    )
+    .unwrap();
+    Balances::mint_into(
+        &sibling_sovereign_account::<Test>(TEMPLATE_PARAID.into()),
+        InitialFund::get(),
+    )
+    .unwrap();
+}
+
+pub fn mock_ext() -> sp_io::TestExternalities {
```

Comment:
why do we need to create a different runtime here? this defeats the purpose of testing our runtime

Reply:
fixed